### PR TITLE
Initial nightly long run implementation

### PR DESCRIPTION
**Pull request by Sushisource**
## What was changed
Add long running nightly test

## Why?
Testing good!

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

**Comment by Sushisource** [comment]
@bergundy I got super sucked in to RxJS land making this work. It's very "fun" but I can't honestly claim I feel at all productive with it lol... at least not yet.

**Comment by Sushisource** [comment]
@bergundy My inclination is to merge this since the nightly runs are working (with the minor exception of figuring out why it "failed" when it looks like it succeeded, need to fix that) and then make any subsequent adjustments (like exposing more configuration options) to the gRPC retry stuff in a later PR. LMK

File: .buildkite/pipeline.yml
Comment (bergundy) [inline .buildkite/pipeline.yml:16]: nit: Can we rename to this?
```suggestion
            - .buildkite/docker/docker-compose-ci.yaml
```

File: packages/client/src/grpc-retry.ts
Comment (bergundy) [inline packages/client/src/grpc-retry.ts:34]: Are these supposed to be `any`?

File: packages/client/src/grpc-retry.ts
Comment (bergundy) [inline packages/client/src/grpc-retry.ts:27]: Can we make this configurable?

File: packages/client/src/grpc-retry.ts
Comment (bergundy) [inline packages/client/src/grpc-retry.ts:11]: I'd rather avoid hardcoding this here, it'd be nicer if this was configurable with a sane default.
We don't have to expose this knob to users, just would make it easier to test.

File: packages/client/src/grpc-retry.ts
Comment (bergundy) [inline packages/client/src/grpc-retry.ts:12]: It's probably far from being an issue but this could be a `Set` for "faster" lookup.

File: packages/test/src/load/setup.ts
Comment (bergundy) [inline packages/test/src/load/setup.ts:10]: Apparently this broke CI, the test we have for checking if a namespace is available is inaccurate.

File: packages/test/src/test-core.ts
Comment (bergundy) [inline packages/test/src/test-core.ts:31]: 🙌

**Review by bergundy — APPROVED** [review]
Had a few minor comments.
Let's merge this once CI is green.

File: packages/client/src/grpc-retry.ts
Comment (Sushisource) [reply packages/client/src/grpc-retry.ts:34]: Yep :(

File: packages/test/src/load/setup.ts
Comment (Sushisource) [reply packages/test/src/load/setup.ts:10]: Well this line didn't, since it's only load and that's running OK - but the thing in `scripts` apparently doesn't work yeah

**Review by bergundy — APPROVED** [review]
Looking at the special handling done for `NoWorkerRegisteredError` think it might be better to postpone deregistering the worker from core (rust) until the final poll results (ShutdownError) have been returned to lang for activities and workflows.

Please don't forget to remove the extra prints before merging.

File: packages/worker/src/worker.ts
Comment (Sushisource) [reply packages/worker/src/worker.ts:1115]: Won't that not matter since it's already the last value in the combined observable of all the pollers/stuff?

Comment (Sushisource) [reply packages/worker/src/worker.ts:1115]: Basically both pollers either already have or are immediately about to encounter the not registered error.

In any case, I agree this would be better handled in core but I didn't want to have to do two merge rounds to wrap up this PR. I will do that in core and then clean this up.

**Comment by Sushisource** [comment]
Merged this even though I'm not 100% sure the "mystery exit code 1" problem is solved on the nightly run. The other stuff is good / worth getting in anyway. I will continue to work on figuring out the weird exit code, just takes ages to repro.

**Pull request merged**
